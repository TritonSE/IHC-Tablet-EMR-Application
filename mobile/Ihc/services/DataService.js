/* global */
// Returns either fakeLocalData or RealData depending on config file
// Config file:
// "FakeDataServices": "true" or "false"
import * as fakeLocalData from '../services/FakeLocalDataService';
import * as localDataService from '../services/LocalDataService';

import * as fakeServerData from '../services/FakeServerDataService';
import * as serverDataService from '../services/ServerDataService';

import config from '../config.json';

// Dont use the Fakes in tests, so that we can test the LocalDataService itself,
// but should stub out its methods for the other tests
let localData = (config.fakeDataServices === 'true') ? fakeLocalData : localDataService;
let serverData = (config.fakeDataServices === 'true') ? fakeServerData : serverDataService;
/*
let localData = (config.fakeDataServices === 'true' || global.__TEST__) ? fakeLocalData : localDataService;
let serverData = (config.fakeDataServices === 'true' || global.__TEST__) ? fakeServerData : serverDataService;
*/
export {localData, serverData};
