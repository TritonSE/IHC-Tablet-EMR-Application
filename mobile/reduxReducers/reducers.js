import {combineReducers} from 'redux';
import * as containerActions from '../reduxActions/containerActions';

// individual reducers have a function name that will equal their property name
// in the state object
// i.e. state.loading will become equal to whatever this function returns
// The state parameter represents the preexisting state, and should be returned
// if the action does not match.
// If the action does match, then return the new value that is passed within the
// action.
function loading(state = false, action) {
  switch (action.type) {
    case containerActions.SET_LOADING:
      return action.loading;
    default:
      return state;
  }
}

function uploading(state = false, action) {
  switch (action.type) {
    case containerActions.IS_UPLOADING:
      return action.uploading;
    default:
      return state;
  }
}

function showRetryButton(state = false, action) {
  switch (action.type) {
    case containerActions.SET_LOADING:
      return action.showRetryButton;
    default:
      return state;
  }
}

function messages(state = {successMessage: null, errorMessage: null}, action) {
  switch (action.type) {
    case containerActions.SET_SUCCESS_MESSAGE:
      return {successMessage: action.message, errorMessage: null};
    case containerActions.SET_ERROR_MESSAGE:
      return {errorMessage: action.message, successMessage: null};
    case containerActions.CLEAR_MESSAGES:
      return {successMessage: null, errorMessage: null};
    default:
      return state;
  }
}

function currentPatientKey(state = null, action) {
  switch (action.type) {
    case containerActions.SET_CURRENT_PATIENT_KEY:
      return action.currentPatientKey;
    default:
      return state;
  }
}

const reducers = combineReducers({
  loading,
  uploading,
  showRetryButton,
  messages,
  currentPatientKey
});

export default reducers;
