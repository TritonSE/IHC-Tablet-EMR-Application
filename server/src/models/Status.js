import mongoose from 'mongoose';

const Schema = mongoose.Schema;
const StatusSchema = Schema({
  patientKey: String,
  name: String,
  birthday: String,
  date: String,
  active: Boolean,
  checkinTime: Number, // timestamp, milliseconds
  triageCompleted: Number, // timestamps for when completed
  doctorCompleted: Number,
  pharmacyCompleted: Number,
  notes: String,
  lastUpdated: Number // timestamp
});
const StatusModel = mongoose.model('Status', StatusSchema);
module.exports = StatusModel;
