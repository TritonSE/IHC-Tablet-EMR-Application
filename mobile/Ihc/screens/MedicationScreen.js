import React, { Component } from 'react';
import {
  ScrollView,
  StyleSheet,
  Text,
  View
} from 'react-native';
import {localData, serverData} from '../services/DataService';
import MedicationTable  from '../components/MedicationTable';
import Container from '../components/Container';
import {stringDate} from '../util/Date';
import DrugUpdate from '../models/DrugUpdate';
import Button from '../components/Button';
import {downstreamSyncWithServer} from '../util/Sync';

export default class MedicationScreen extends Component<{}> {
  /*
   * Props:
   * name: patient's name for convenience
   * patientKey: string of patient's key
   */
  constructor(props) {
    super(props);

    this.state = {
      loading: false,
      showRetryButton: false,
      updates: [],
      errorMsg: null,
      successMsg: null,
      medicationCheckmarks: [],
      todayDate: stringDate(new Date()),
      upstreamSyncing: false, // Should be set before server calls to declare what kind of syncing
    };
    this.props.navigator.setOnNavigatorEvent(this.onNavigatorEvent.bind(this));
  }

  // Reload table after new medication updates
  // Replaces componentDidMount() because this will be called around the same
  // time
  onNavigatorEvent(event) {
    if (event.id === 'willAppear') {
      this.syncAndLoadMedications();
    }
  }

  refillMedication = (prevDrugUpdate) => {
    const newUpdate = Object.assign({}, prevDrugUpdate);
    newUpdate.date = this.state.todayDate;

    try {
      localData.updateDrugUpdate(newUpdate);
    } catch(e) {
      this.setState({errorMsg: e.message});
      return;
    }

    this.setState({loading: true, upstreamSyncing: true});
    serverData.updateDrugUpdate(newUpdate)
      .then( () => {
        // View README: Handle syncing the tablet, point 3 for explanation
        if(this.state.loading) {
          // if successful, then reload data
          this.syncAndLoadMedications();

          this.setState({
            loading: false,
            showRetryButton: false,
            successMsg: 'Saved successfully',
            errorMsg: null
          });
        }
      })
      .catch( (e) => {
        if(this.state.loading) {
          localData.markPatientNeedToUpload(this.props.patientKey);
          this.setState({
            errorMsg: e.message,
            successMsg: null,
            loading: false,
            showRetryButton: true
          });
        }
      });
  }

  changeMedication = (prevDrugUpdate) => {
    this.props.navigator.push({
      screen: 'Ihc.MedicationUpdateScreen',
      title: 'Medication',
      passProps: {
        drugUpdate: prevDrugUpdate,
        action: 'change',
        patientKey: this.props.patientKey,
        name: this.props.name
      }
    });
  }

  discontinueMedication = (prevDrugUpdate) => {
    // Add a dummy drug update to updates prop
    const newUpdate = DrugUpdate.getDiscontinueDummy(prevDrugUpdate);
    this.setState({
      updates: this.state.updates.concat([newUpdate])
    });
  }

  createNewMedication = () => {
    this.props.navigator.push({
      screen: 'Ihc.MedicationUpdateScreen',
      title: 'Medication',
      passProps: {
        action: 'new',
        patientKey: this.props.patientKey,
        name: this.props.name
      }
    });
  }

  syncAndLoadMedications = () => {
    this.setState({ loading: true, upstreamSyncing: false, errorMsg: null, successMsg: null });

    // Load local data in beginning to display even if sync doesn't work
    let updates = localData.getMedicationUpdates(this.props.patientKey);
    let statusObj = localData.getStatus(this.props.patientKey, this.state.todayDate);
    const checkmarks = statusObj.medicationCheckmarks;
    this.setState({
      updates: updates,
      medicationCheckmarks: checkmarks,
    });

    downstreamSyncWithServer()
      .then((failedPatientKeys) => {
        if(failedPatientKeys.length > 0) {
          throw new Error(`${failedPatientKeys.length} patients didn't properly sync.`);
        }

        let updates = localData.getMedicationUpdates(this.props.patientKey);
        let statusObj = localData.getStatus(this.props.patientKey, this.state.todayDate);
        const checkmarks = statusObj.medicationCheckmarks;
        this.setState({
          updates: updates,
          medicationCheckmarks: checkmarks,
          loading: false
        });
      })
      .catch(err => {
        this.setState({loading: false, errorMsg: err.message});
      });
  }

  saveCheckmarks = () => {
    // Local checkmark saves are handled in MedicationTable directly.
    const statusObj = localData.getStatus(this.props.patientKey, this.state.todayDate);

    this.setState({
      loading: true,
      errorMsg: null,
      successMsg: null,
    });

    serverData.updateStatus(statusObj)
      .then( () => {
        // View README: Handle syncing the tablet, point 3 for explanation
        if(this.state.loading) {
          this.setState({
            successMsg: 'Saved',
            errorMsg: null,
            loading: false,
            showRetryButton: false
          });
        }
      })
      .catch( (e) => {
        if(this.state.loading) {
          localData.markPatientNeedToUpload(this.props.patientKey);
          this.setState({
            successMsg: null,
            errorMsg: e.message,
            loading: false,
            showRetryButton: true
          });
        }
      });
  }

  // station: 'Doctor' or 'Pharmacy'
  updateStatus(station) {
    this.setState({
      loading: true,
      errorMsg: null,
      successMsg: null,
    });

    let statusObj = {};
    if(station !== 'Doctor' && station !== 'Pharmacy') {
      throw new Error(`Received invalid station: ${station}`);
    }

    const fieldName = station === 'Doctor' ? 'doctorCompleted' : 'pharmacyCompleted';
    try {
      statusObj = localData.updateStatus(this.props.patientKey, this.state.todayDate,
        fieldName, new Date().getTime());
    } catch(e) {
      this.setState({errorMsg: e.message, successMsg: null, loading: false});
      return;
    }

    serverData.updateStatus(statusObj)
      .then( () => {
        // View README: Handle syncing the tablet, point 3 for explanation
        if(this.state.loading) {
          this.setState({
            successMsg: `${station} marked as completed`,
            errorMsg: null,
            loading: false,
            showRetryButton: false
          });
        }
      })
      .catch( (e) => {
        if(this.state.loading) {
          localData.markPatientNeedToUpload(this.props.patientKey);
          this.setState({
            successMsg: null,
            errorMsg: e.message,
            loading: false,
            showRetryButton: true
          });
        }
      });
  }

  completed = () => {
    this.updateStatus('Doctor');
  }

  filled = () => {
    this.updateStatus('Pharmacy');
  }

  // If Loading was canceled, we want to show a retry button
  setLoading = (val, canceled) => {
    let errorMsg = null;
    // View README: Handle syncing the tablet, point 5 for explanation
    if(canceled && this.state.upstreamSyncing === false) {
      errorMsg = 'Canceling may cause data to be out of sync.';
    }
    this.setState({loading: val, showRetryButton: canceled, errorMsg: errorMsg});
  }

  setMsg = (type, msg) => {
    const obj = {};
    obj[type] = msg;
    const other = type === 'successMsg' ? 'errorMsg' : 'successMsg';
    obj[other] = null;
    this.setState(obj);
  }

  render() {
    return (
      <Container loading={this.state.loading}
        errorMsg={this.state.errorMsg}
        successMsg={this.state.successMsg}
        setLoading={this.setLoading}
        setMsg={this.setMsg}
        patientKey={this.props.patientKey}
        showRetryButton={this.state.showRetryButton}
      >

        <View style={styles.headerContainer}>
          <Text style={styles.title}>
            {this.props.name}'s Medications
          </Text>

          <Text>R: Refill, D: Change Dose, X: Cancel</Text>
          <Text>T: Taking, N: Not Taking, I: Taking incorrectly</Text>
        </View>

        <ScrollView contentContainerStyle={styles.tableContainer} horizontal>
          <MedicationTable style={styles.table}
            refill={this.refillMedication}
            change={this.changeMedication}
            discontinue={this.discontinueMedication}
            updates={this.state.updates}
            medicationCheckmarks={this.state.medicationCheckmarks}
            patientKey={this.props.patientKey}
          />
        </ScrollView>

        <View style={styles.footerContainer}>
          <Button
            onPress={this.saveCheckmarks}
            style={styles.button}
            text='Save checkmarks'>
          </Button>

          <Button
            onPress={this.createNewMedication}
            style={styles.button}
            text='New Medication'>
          </Button>

          <Button
            onPress={this.completed}
            style={styles.button}
            text='Completed Prescribing'>
          </Button>

          <Button
            onPress={this.filled}
            style={styles.button}
            text='Filled'>
          </Button>
        </View>
      </Container>
    );
  }
}

const styles = StyleSheet.create({
  headerContainer: {
    justifyContent: 'center',
    alignItems: 'center',
  },
  title: {
    fontSize: 20,
    textAlign: 'center',
  },
  tableContainer: {
    flex: 0,
  },
  footerContainer: {
    flex: 1,
    flexDirection: 'row',
    height: 40,
    margin: 4,
  },
  button: {
    width: 120,
    height: 60 
  }
});
