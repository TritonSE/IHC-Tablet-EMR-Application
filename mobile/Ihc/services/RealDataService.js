// This file should hold all the fetch() calls to the Express server
// and the local database calls to the Realm DB
// TODO: Keep realm in sync with mongo
// All lastUpdated fields on mobile-side should be handled within this file.

import Patient from '../models/Patient';
import Status from '../models/Status';
import Soap from '../models/Soap';
import Triage from '../models/Triage';
import DrugUpdate from '../models/DrugUpdate';
import Settings from '../models/Settings';

import {createPatientHelper} from './functions/CreatePatient';
import {signinPatientHelper} from './functions/SigninPatient';
import {updateStatusHelper} from './functions/UpdateStatus';
import {updateSoapHelper} from './functions/UpdateSoap';
import {createDrugUpdateHelper} from './functions/CreateDrugUpdate';
import {getSoapHelper} from './functions/GetSoap';
import {getTriageHelper} from './functions/GetTriage';
import {getPatientHelper} from './functions/GetPatient';
import {getPatientSelectRowsHelper} from './functions/GetPatientSelectRows';
import {updateTriageHelper} from './functions/UpdateTriage';
import {getMedicationUpdatesHelper} from './functions/GetMedicationUpdates';
import {uploadUpdatesHelper} from './functions/UploadUpdates';
import {downloadUpdatesHelper} from './functions/DownloadUpdates';

import Realm from 'realm';
import {stringDate} from '../util/Date';
import config from '../config.json';

// Must set the fetchUrl to the server's IP Address and Port
const fetchUrl = config.fetchUrl;

const realm = new Realm({
  schema: [Patient, Status, Soap, Triage, DrugUpdate, Settings],
  deleteRealmIfMigrationNeeded: true, // TODO: delete when done with dev
});

export function createPatient(patient) {
  return createPatientHelper(patient, realm, fetchUrl);
}

// Check that patient exists, and if so then create a status object for them
export function signinPatient(patientForm) {
  return signinPatientHelper(patientForm, realm, fetchUrl);
}

// field: Any of fields from Status.schema
// value: should match the type that the Status.schema says
export function updateStatus(patientKey, strDate, field, value) {
  return updateStatusHelper(patientKey, strDate, field, value, realm, fetchUrl);
}

export function createDrugUpdate(update) {
  return createDrugUpdateHelper(update, realm, fetchUrl);
}

export function updateSoap(update) {
  return updateSoapHelper(update, realm, fetchUrl);
}

// Returns SOAP object if it exists, or undefined if not
export function getSoap(patientKey, strDate) {
  return getSoapHelper(patientKey, strDate, realm, fetchUrl);
}

export function updateTriage(update) {
  return updateTriageHelper(update, realm, fetchUrl);
}

// Returns Triage object if it exists, or undefined if not
export function getTriage(patientKey, strDate) {
  return getTriageHelper(patientKey, strDate, realm, fetchUrl);
}

export function getPatient(patientKey) {
  return getPatientHelper(patientKey, realm, fetchUrl);
}

/*
 * Return the statuses of the patients that are active and for this date
 */
export function getPatientSelectRows() {
  return getPatientSelectRowsHelper(realm, fetchUrl);
}

export function getMedicationUpdates(patientKey) {
  return getMedicationUpdatesHelper(patientKey, realm, fetchUrl);
}

export function uploadUpdates() {
  return uploadUpdatesHelper(realm, fetchUrl);
}

export function downloadUpdates() {
  return downloadUpdatesHelper(realm, fetchUrl);
}
