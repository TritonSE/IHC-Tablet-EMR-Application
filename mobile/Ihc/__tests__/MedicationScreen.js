import 'react-native';
import React from 'react';
import MedicationScreen from '../screens/MedicationScreen';

import DrugUpdate from '../models/DrugUpdate';
import Status from '../models/Status';
import sinon from 'sinon';
import {localData} from '../services/DataService';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

import { createStore } from 'redux';
import reducers from '../reduxReducers/reducers';
import { Provider } from 'react-redux';

const store = createStore(reducers);

it('renders correctly', () => {
  sinon.useFakeTimers(100);
  const drug1 = DrugUpdate.getInstance();
  const drug2 = DrugUpdate.getInstance();
  sinon.stub(localData, 'getMedicationUpdates').returns([drug1, drug2]);
  sinon.stub(localData, 'getStatus').returns(Status.newStatus({key: 'key'}));

  const fakeNavigator = { setOnNavigatorEvent: (param) => {} };
  const json = renderer.create(
    <Provider store={store}>
      <MedicationScreen navigator={fakeNavigator} />
    </Provider>
  ).toJSON();
  expect(json).toMatchSnapshot();
});
