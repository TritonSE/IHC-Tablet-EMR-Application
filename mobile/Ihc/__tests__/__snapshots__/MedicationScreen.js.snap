// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F5FCFF",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "flex": 0,
        "justifyContent": "center",
        "padding": 20,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 20,
              "textAlign": "center",
            }
          }
        >
          's Medications
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          R: Refill, D: Change Dose, X: Cancel
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          T: Taking, N: Not Taking, I: Taking incorrectly
        </Text>
      </View>
      <RCTScrollView
        contentContainerStyle={
          Object {
            "flex": 0,
          }
        }
        horizontal={true}
      >
        <View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              No data to show
            </Text>
          </View>
        </View>
      </RCTScrollView>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "height": 40,
            "margin": 4,
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#2196F3",
              "borderRadius": 2,
              "elevation": 4,
              "height": 60,
              "justifyContent": "center",
              "margin": 4,
              "opacity": 1,
              "padding": 8,
              "width": 120,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#fefefe",
                  "fontWeight": "500",
                  "textAlign": "center",
                },
                undefined,
              ]
            }
          >
            Save checkmarks
          </Text>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#2196F3",
              "borderRadius": 2,
              "elevation": 4,
              "height": 60,
              "justifyContent": "center",
              "margin": 4,
              "opacity": 1,
              "padding": 8,
              "width": 120,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#fefefe",
                  "fontWeight": "500",
                  "textAlign": "center",
                },
                undefined,
              ]
            }
          >
            New Medication
          </Text>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#2196F3",
              "borderRadius": 2,
              "elevation": 4,
              "height": 60,
              "justifyContent": "center",
              "margin": 4,
              "opacity": 1,
              "padding": 8,
              "width": 120,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#fefefe",
                  "fontWeight": "500",
                  "textAlign": "center",
                },
                undefined,
              ]
            }
          >
            Completed Prescribing
          </Text>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#2196F3",
              "borderRadius": 2,
              "elevation": 4,
              "height": 60,
              "justifyContent": "center",
              "margin": 4,
              "opacity": 1,
              "padding": 8,
              "width": 120,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#fefefe",
                  "fontWeight": "500",
                  "textAlign": "center",
                },
                undefined,
              ]
            }
          >
            Filled
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#F5FCFF",
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "red",
          "textAlign": "center",
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "green",
          "textAlign": "center",
        }
      }
    />
  </View>
</View>
`;
